runtime/src/main/cpp/Memory.cpp

```diff
@@ -1069,8 +1070,8 @@ ALWAYS_INLINE void runDeallocationHooks(ContainerHeader* container) {
   ObjHeader* obj = reinterpret_cast<ObjHeader*>(container + 1);
   for (uint32_t index = 0; index < container->objectCount(); index++) {
     auto* type_info = obj->type_info();
-    if (type_info == theWorkerBoundReferenceTypeInfo) {
-      DisposeWorkerBoundReference(obj);
+    if (type_info == theCleanerImplTypeInfo) {
+        DisposeCleaner(obj);
```

Comment:
Running Kotlin code (`Kotlin_CleanerImpl_clean`) during GC is not great (see e.g. recent recursive GC issue).

Reply:
Yep, moved scheduling to C++